### ISSUE TITLE
Crash when copy/move

### ISSUE THREAD

**Issue by dalvand** [vifm/vifm#761]
Vifm keeps crashing when copying files and folders with `Segmentation fault (core dumped)` error. Where does it capture any logs to see what's going on?

I build the latest version of vifm in Ubuntu 20.04.

journalctl shows this for every crash:

`kernel: vifm[464888]: segfault at 0 ip 00007f52c8ead4e0 sp 00007ffc0f99c3c8 error 4 in libc-2.31.so[7f52c8d4b000+178000]`

**Comment by mark2185** [comment]
Have you tried toggling `syscalls` with `set syscalls`?

```
                                               *vifm-'syscalls'*
syscalls
type: boolean
default: false

When disabled, vifm will rely on external applications to perform file-system
operations, otherwise system calls are used instead (much faster and supports
progress tracking).  The option should eventually be removed.  Mostly
*nix-like systems are affected.
```

**Comment by dalvand** [comment]
Yes, `set syscalls` is in my vifmrc by default and I did not change it.

**Comment by mark2185** [comment]
In the meantime, try `set nosyscalls` to see if that alleviates the problem.

**Comment by dalvand** [comment]
I will try that when this happened again. But, what is your interpretation from the error above related to `libc-2.31.so`? 

I also experienced several errors during copy/move operations like this one:
https://ibb.co/bbJKSSc

I tried the same operations with other file managers and I didn't get these errors.

**Comment by mark2185** [comment]
> But, what is your interpretation from the error above related to libc-2.31.so?

No idea, I'm just a random user of the program, I'm not familiar with the codebase. Might be a bug.

> I also experienced several errors during copy/move operations like this one:

Do the errors always happen when copying to the external disk? (I presume the destination in the screenshot is a removable device)
Does it happen only with the encrypted disk?

**Comment by dalvand** [comment]
I just recently switched to vifm and haven't tried it in different scenarios. These specific crashes happened with external disk. But, I have seen errors during copy/move with internal operations as well. 

Before reporting the issues here, I tried the same operation with other file managers and I didn't experienced any crash or error for the same operations.

**Comment by xaizek** [comment]
Log is in `$VIFM/log` or in `~/.local/share/vifm/log`, not sure it will contain many details, but take a look after specifying `--logging`.

Since you've built Vifm yourself, take a look at instructions on [reporting crashes](https://wiki.vifm.info/index.php/How_to_report_a_crash_in_a_most_useful_way_(with_a_backtrace)).

**Comment by dalvand** [comment]
Do you think if this is related or caused by [#763](https://github.com/vifm/vifm/issues/763)?

**Comment by xaizek** [comment]
Not sure it would cause a crash. By the way, you haven't specified conditions for the crash:
 * <kbd>p</kbd>/<kbd>P</kbd>?
 * `:copy`/`:move`?
 * single/multiple files or both?

**Comment by dalvand** [comment]
I use the following two commands for copy and move:
`:yank | winrun , put`
`:yank | winrun , put!`

The crashes happened when I was copying or moving multiple files and folders to an external hard drive.
